### PR TITLE
added fortran for n1 and python wrapper in solenspipe

### PR DESCRIPTION
**Pull request by qujia7**

### CODE REVIEW

**Review by msyriac — CHANGES_REQUESTED** [review]
This looks great as a start! We may want to follow the example of simonsobs/pixell for wrapping in Fortran code to make this a bit more flexible. Let me know if you can look at that. Also looks like a lot of n1 derivative stuff is already here!

File: bin/Makefile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        F90FLAGS = "-fopenmp"
+        OPT = --opt=-O3 -lgfortran -lifcore
+else ifeq (${NERSC_HOST}, cori)
+        FF = gfortran
+        FPY = f2py
+        F90FLAGS = "-fopenmp"
+        OPT = --opt=-O3 -lgfortran -lifcore
+else
+        FF = gfortran
+        FPY = f2py-2.7
```
Comment (msyriac) [inline bin/Makefile]: We should not write any code that references anything related to Python 2.

**Comment by msyriac** [comment]
I've added the fortran compiling to setup.py (the way that pixell does it). The Makefile is not needed anymore. One just needs to run e.g.:
python setup.py build_ext -i
or e.g.:
python setup.py build_ext -i --fcompiler=intelem --compiler=intelem

**Comment by ajvanengelen** [comment]
This looks great so far - thanks!

**Comment by msyriac** [comment]
@qujia7 can we modify this so that the fortran function passes back the result without having to save to disk? (Saving stuff with the same file name to disk is not thread-safe!)

**Comment by msyriac** [comment]
Similarly, ideally, the lens potential and CMB spectra should be passed as arrays, not as filenames.

**Comment by msyriac** [comment]
@qujia7 would be good to have the L_step as an argument too

**Review by msyriac — APPROVED** [review]

**Pull request merged**